spark/src/main/java/org/apache/iceberg/actions/Actions.java

```diff
@@ -77,6 +82,81 @@ public ExpireSnapshotsAction expireSnapshots() {
     return new ExpireSnapshotsAction(spark, table);
   }
 
+  /**
+   * Converts the provided table into an Iceberg table in place. The table will no longer be represented by it's
+   * previous provider in the session catalog and a new metadata directory will be created at the table's location.
+   *
+   * @param tableName Table to be converted
+   * @return Action to perform migration
+   */
+  public static CreateAction migrate(String tableName) {
+    try {
+      return DynMethods.builder("migrate")
+          .impl(implClass(), String.class).buildStaticChecked()
+          .invoke(tableName);
+    } catch (NoSuchMethodException ex) {
+      throw new UnsupportedOperationException("Snapshot is not implemented for this version of Spark");
+    }
+  }
+
+  /**
+   * Converts the provided table into an Iceberg table in place. The table will no longer be represented by it's
+   * previous provider in the session catalog and a new metadata directory will be created at the table's location.
+   *
+   * @param tableName Table to be converted
+   * @param spark     Spark session to use for looking up table
+   * @return Action to perform migration
+   */
+  public static CreateAction migrate(SparkSession spark, String tableName) {
+    try {
+      return DynMethods.builder("migrate")
+          .impl(implClass(), SparkSession.class, String.class).buildStaticChecked()
+          .invoke(spark, tableName);
+    } catch (NoSuchMethodException ex) {
+      throw new UnsupportedOperationException("Snapshot is not implemented for this version of Spark");
+    }
+  }
+
+  /**
+   * Creates an independent Iceberg snapshot of a given table. The Snapshot can be altered, appended or deleted without
+   * causing any change to the original table the snapshot is based on. New data and metadata will be created in the
+   * location passed to this method.
+   *
+   * @param sourceId Snapshot's source data
+   * @param destId   Name of the new snapshot
```

Comment:
The wording in this Javadoc and the argument names seem a little confusing to me. I think it should be "Creates a new Iceberg table that is a snapshot of the given source table", then "The new table can be altered, . . .". Referring to the table that gets created as a table should help users understand what is happening a bit better than referring to it as a "snapshot", which has a conflicting meaning when used as a noun. Here, I think we want to stick to using it as a verb.

Since the "source" and "dest" are tables, I'd use "sourceName" or "sourceTable" and "destName" or "destTable".